packages/amplify-graphql-model-transformer/src/resources/amplify-dynamodb-table/amplify-table-manager-lambda/amplify-table-manager-handler.ts

```diff
@@ -806,7 +806,7 @@ export const getSseUpdate = (currentState: TableDescription, endState: CustomDDB
  * @returns Boolean indicating if the tags are updated
  */
 export const requiresTagsUpdate = (currentTags: DynamoDBTag[], newTags?: DynamoDBTag[]): boolean => {
-  if (!newTags || newTags.length === 0) {
```

Comment:
Calling the UpdateTags with an empty array throws an runtime error. The PR is intended to fix Gen2 workflow which always contains at least one tag. If we need to handle CDK deployments as well, we need to handle it separately, this logic won't work.

Reply:
I have added a logic to remove the existing tags. It required additional IAM permission `UntagResource` and new API call.